metagpt/actions/write_docstring.py

```diff
@@ -0,0 +1,191 @@
+import ast
+import contextlib
+from typing import Literal
+
+from metagpt.actions.action import Action
+from metagpt.utils.common import OutputParser
+from metagpt.utils.pycst import merge_docstring
+
+PYTHON_DOCSTRING_SYSTEM = '''### Requirements
+1. Add docstrings to the given code following the {style} style.
+2. Replace the function body with an Ellipsis object(...) to reduce output.
+3. If the types are already annotated, there is no need to include them in the docstring.
+4. Extract only class, function or the docstrings for the module parts from the given Python code, avoiding any other text.
+
+### Input Example
+```python
+def function_with_pep484_type_annotations(param1: int) -> bool:
+    return isinstanc(param1, int)
+
+class ExampleError(Exception):
+    def __init__(self, msg: str):
+        self.msg = msg
+```
+
+### Output Example
+```python{example}```
```

Comment:
python
{example}
?

Reply:
The example is one of `PYTHON_DOCSTRING_EXAMPLE_GOOGLE`, `PYTHON_DOCSTRING_EXAMPLE_NUMPY`, or `PYTHON_DOCSTRING_EXAMPLE_SPHINX` code snippets, which depend on the style you use. These code snippets all begin and end with `\n`.

Reply:
although example snippets seperate with `\n`, the `python\n{example}` format is more clear.

Reply:
OK, I will remove the newline characters from the example snippets.